<launch>
  <arg name="paused" default="false"/>
  <arg name="debug" default="false"/>
  <arg name="gui" default="true" />
  <arg name="headless" default="false"/>
  <arg name="use_ground_truth" default="true" />
  <arg name="real_machine" default="false" />
  <arg name="simulation" default="true" />
  <arg name="control_mode" default="0" />
  <arg name="estimate_mode" default="2" />
  <arg name="uav_name" default="hawk" />
  <arg name="start" default="true" />

  <include file="$(find mbzirc2020_task1_common)/launch/mbzirc_arena_1.launch" >
    <arg name="paused" default="$(arg paused)"/>
    <arg name="debug" default="$(arg debug)"/>
    <arg name="gui" value="$(arg gui)" />
    <arg name="headless" default="$(arg headless)"/>
    <arg name="use_ground_truth" default="$(arg use_ground_truth)"/>
    <arg name="uav_name" value="$(arg uav_name)" />
    <arg name="start" value="$(arg start)" />
  </include>

  <!-- Spawn simulated HYDRUS uav -->
  <include file="$(find hydrus)/launch/bringup.launch" >
    <arg name="real_machine" value="$(arg real_machine)" />
    <arg name="simulation" value="$(arg simulation)" />
    <arg name="control_mode" value="$(arg control_mode)" />
    <arg name="estimate_mode"  value= "$(arg estimate_mode)" />
    <arg name="type" value="quad" />
    <arg name="headless" value="true" />
    <arg name="launch_gazebo" value="false" />
  </include>

  <group unless="$(arg headless)" >
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find mbzirc2020_task1_tasks)/config/task1.rviz"/>
  </group>


</launch>
